v22.1/manage-certs-vault.md

```diff
@@ -0,0 +1,660 @@
+    ~~~
+
+1. Use `openssl` to examine a certificate, confirming that it has been generated and copied correctly.
+    {% include_cached copy-clipboard.html %}
+    ```shell
+    openssl x509 -in "${secrets_dir}/${node1name}/cockroach_cluster_ca.crt" -text | less
+    ```
+    ~~~txt
+    Certificate:
+        Data:
+            Version: 3 (0x2)
+            Serial Number:
+                1a:72:ac:49:e9:38:38:65:e3:40:16:a8:48:6e:34:a0:3f:0f:00:96
+        Signature Algorithm: sha256WithRSAEncryption
+            Issuer:
+            Validity
+                Not Before: Jun 21 17:24:39 2022 GMT
+                Not After : Jul 23 17:25:09 2022 GMT
+            Subject:
+            Subject Public Key Info:
+                Public Key Algorithm: rsaEncryption
+                    Public-Key: (2048 bit)
+          ...
+    ~~~
+
+    
+### Create PKI roles and issue certificates for the nodes
+
+In Vault, a PKI role is a template for a certificate.
+
+1.  Create a node role for each node. The role will be used to generate certifates for the corresponding node.
+
+    Each certificate is tailored to the node:
+    - The extended key usages attribute `ext_key_usage` must include both server and client auth usages; this is because nodes must frequently initiate requests to other nodes in order to maintain cluster synchrony and load-balance work.
+    - The Subject Alternative Name (SAN) - IP addresses field contains:
+      - the IP address of the node on the internal network of your GCP project (so the node can serve at that address  locally, to other nodes).
+      - the IP address of your cluster on the external, public internet (so the node can serve at that address publicly, to application servers).
+
+    {{site.data.alerts.callout_info}}
+    Note that certificate attributes must be provided with Vault's specific parameter syntax, which is documented here: 
+    {{site.data.alerts.end}}
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    vault write "cockroach_cluster_ca/roles/${node1name}" \
+    allow_bare_domains=true \
+    common_name="node" \
+    allowed_domains="${node1name},localhost,node" \
+    ip_sans="${node1addr},${ex_ip}" \
+    ext_key_usage="server_auth,client_auth" \
+    max_ttl=1000h # just over a month
+
+    vault write "cockroach_cluster_ca/roles/${node2name}" \
+    allow_bare_domains=true \
+    common_name="node" \
+    allowed_domains="${node2name},localhost,node" \
+    ip_sans="${node2addr},${ex_ip}" \
+    ext_key_usage="server_auth,client_auth" \
+    max_ttl=1000h # just over a month 
+
+    vault write "cockroach_cluster_ca/roles/${node3name}" \
+    allow_bare_domains=true \
+    common_name="node" \
+    allowed_domains="${node3name},localhost,node" \
+    ip_sans="${node3addr},${ex_ip}" \
+    ext_key_usage="server_auth,client_auth" \
+    max_ttl=1000h # just over a month 
+    ~~~
+
+    ~~~txt
+    Success! Data written to: cockroach_node_ca/roles/cockroach-cluster-alpha-1bth
```

Comment:
Maybe use `${node1name}` and so forth in these success messages to reinforce that they will see one success message for each node?